src/org/opensolaris/opengrok/history/GitRepository.java

```diff
@@ -170,11 +125,48 @@ Executor getHistoryLogExecutor(final File file, String sinceRevision)
         cmd.add("--pretty=fuller");
         cmd.add("--date=rfc");
 
+        if (file.isFile() && RuntimeEnvironment.getInstance().isHandleHistoryOfRenamedFiles()) {
+            cmd.add("--follow");
+        }
+
+        if (sinceRevision != null) {
+            cmd.add(sinceRevision + "..");
+        }
+
+        if (filename.length() > 0) {
+            cmd.add(filename);
+        }
+
+        return new Executor(cmd, new File(getDirectoryName()), sinceRevision != null);
+    }
+
+    Executor getRenamedFilesExecutor(final File file, String sinceRevision) throws IOException {
+        String abs = file.getCanonicalPath();
+        String filename = "";
+        if (abs.length() > directoryName.length()) {
+            filename = abs.substring(directoryName.length() + 1);
+        }
+
+        ensureCommand(CMD_PROPERTY_KEY, CMD_FALLBACK);
+        List<String> cmd = new ArrayList<>();
+        cmd.add(RepoCommand);
+        cmd.add("log");
+        cmd.add("-M8"); // similarity 80%
```

Comment:
The default value is 50% so I pushed that up to 80%. 100% is to much I think...one can change the location of the file while changing some lines